coalib/processes/Processing.py

```diff
@@ -459,13 +460,15 @@ def yield_ignore_ranges(file_dict):
                                    1,
                                    line_number,
                                    len(file[line_number-1])))
-                elif 'ignore ' in line:
-                    end_line = min(line_number + 1, len(file))
-                    yield (get_ignore_scope(line, 'ignore '),
-                           SourceRange.from_values(
-                               filename,
-                               line_number, 1,
-                               end_line, len(file[end_line - 1])))
+
+                for ignore_stmt in ['ignore ', 'noqa ', 'noqa']:
```

Comment:
before it was 
```
                elif 'ignore ' in line:
                    end_line = min(line_number + 1, len(file))
                    yield (get_ignore_scope(line, 'ignore '),
                           SourceRange.from_values(
                               filename,
                               line_number, 1,
                               end_line, len(file[end_line - 1])))
```
so the `elif 'ignore ' in line` was only evaluated if the previous `if ...` failed. I would keep this code flow.

So
> Am I supposed to be doing
> else: for ....

yes :)

> That doesn't work without the code breaking because the for loop requires the if statement for 
> completion. :(

I don't understand that :O